### ISSUE TITLE
Snapshots are not read properly within one container

### ISSUE THREAD

**Issue by A3a3e1** [elasticio/sailor-jvm#94]
Snapshots seem to be writing correctly but reading of the snapshot is not correct.
Executing messages within one container always gets the old snapshot value. Executing a new message after a container slept, got a correct snapshot value.

Steps to reproduce:
1. Build a flow where you have the possibility to read the snapshots value in the trigger (JAVA-based) component (you can ask me, I have a ready-to-use debug component for this purpose).
2. Run a flow for the first time and see the logs like:
```
Old Snapshot: {}
...
Emitting new snapshot{\"time\":\"2024-05-07T13:16:46.075Z\"}
```
3. Run a flow for the second time within a short period to not the container go to sleep and see the logs like:
```
Old Snapshot: {}
...
Emitting new snapshot{\"time\":\"2024-05-07T13:17:00.075Z\"}
```
4. So far, we see that the old snapshot value was read, not the new one
5. Wait a few minutes until the container will not be in 'Sleeping' mode.
6. Run a flow for the third time and see the logs like:
```
Old Snapshot: {\"time\":\"2024-05-07T13:17:00.075Z\"}
...
Emitting new snapshot{\"time\":\"2024-05-07T13:17:01.000Z\"}
```
7. Here we can see that the snapshot value from the previous execution has been written correctly and read successfully after the container restarted.
8. Optionally, you can execute the flow again wile the container is still up and see the same behavior when the old snapshot will be read, not the new one.

It is only an assumption, that the issue is with reading the snapshot. Needs deeper investigation. Ths issue might also be in an incorrect snapshot write.

**Comment by A3a3e1** [comment]
It seems that this is not a bug, but rather a feature. Snapshots are considered as a Step in the Java Sailor - a set of fields that are singleton (like credentials) and are loaded once per container. 
While the message body (message, headers, attachments) is loaded for each execution, the snapshots are loaded once per container.
https://github.com/elasticio/sailor-jvm/blob/20b8022a32d8dc254ba0545e8c244b581864b74e/src/main/java/io/elastic/sailor/impl/MessageConsumer.java#L128

**Comment by HannaTrotsenko** [comment]
New sailor version is released 4.0.2

**Comment by MykytaDuniashev** [comment]
I notified Infobip in the Slack [thread](https://elasticio.slack.com/archives/C01FLMQ0PB5/p1713268714629679)
